### PR TITLE
feat: Allow the user to sign up with OAuth

### PR DESCRIPTION
**Pull request by WindSpiritSR**
Allow the user to sign up when he tries to use OAuth to login but there's no account linked to the OAuth

Deployed at: https://casdoor.leviatan.cn/login/oauth/authorize?client_id=00c5f3410de2ebf737e4&response_type=code&redirect_uri=https://www.leviatan.cn&scope=read&state=casdoor

I closed #102 by an incorrect operation, I fixed problems at this PR.

1. Reuesd signup page
2. I don't know why, but my email and SMS provider config doesn't work, so I closed the Verification Code check only on this test server, you can enter any character in the `email code` and `phone code` to pass the verification.
3. Reformat by IDE

image

Signed-off-by: WindSpiritSR simon343riley@gmail.com

### CODE REVIEW

**Comment by hsluoyz** [comment]
@Kininaru plz review.

**Comment by hsluoyz** [comment]
@WindSpiritSR 

image

**Review by Kininaru — COMMENTED** [review]
1. Can not open demo site
2. Please turn off auto formatting for the front end, too many lines changed...

**Comment by WindSpiritSR** [comment]
@hsluoyz Deployed at: https://casdoor.leviatan.cn/login/oauth/authorize?client_id=b2f633c65937f2b478f8&response_type=code&redirect_uri=https://www.leviatan.cn/&scope=read&state=casdoor

**Comment by Kininaru** [comment]
@WindSpiritSR redirect uri seems not right

File: controllers/auth.go [hunk 1]
```diff
@@ -263,16 +259,15 @@ func (c *ApiController) Login() {
 
 				if !providerItem.CanSignUp {
 					resp = &Response{Status: "error", Msg: fmt.Sprintf("The account for provider: %s and username: %s (%s) does not exist and is not allowed to sign up as new account via %s, please use another way to sign up", provider.Type, userInfo.Username, userInfo.DisplayName, provider.Type)}
-					c.Data["json"] = resp
-					c.ServeJSON()
-					return
```
[the remark this answers is not in the document]
Comment (WindSpiritSR) [reply controllers/auth.go]: my mistake, fixed

**Pull request closed without merging**